storage/database/src/backend/heed/env.rs

```diff
@@ -268,12 +264,16 @@ where
         tx_ro: &heed::RoTxn<'env>,
     ) -> Result<impl DatabaseRo<T> + DatabaseIter<T>, RuntimeError> {
         // Open up a read-only database using our table's const metadata.
-        Ok(HeedTableRo {
-            db: self
-                .open_database(tx_ro, Some(T::NAME))?
-                .ok_or(RuntimeError::TableNotFound)?,
-            tx_ro,
-        })
+        //
+        // INVARIANT:
+        // We only care about setting the flags and/or comparison functions
+        // on write tables, read tables can't/don't add data so it doesn't matter.
+        // The sorted data from write tables will appear for read tables regardless.
```

Comment:
Uhmm is this defiantly correct? 

How is LMDB supposed to search the data correctly without the correct comparison function?

Reply:
Ah ok, LMDB caches the comparison function, that makes sense, thanks.

Users would be required to create these tables/ open them rw on startup for these flags to be set (I forgot if that is already a requirement or if we dropped that), with your other comment though this is already needed.

Reply:
I would update the comment to include that LMDB is caching the comparison function